extension/controller/backgroundScript.js

```diff
@@ -322,6 +336,15 @@ const messageListener = function(message, sender) {
             translatingAsBrowse: message.translatingAsBrowse
           });
           break;
+        case "errorCollectionConsent":
+          Sentry.getCurrentHub().getClient()
+            .getOptions().enabled = message.consent;
+          browser.storage.local.set({ errorCollectionConsent: message.consent });
+          break;
+        case "telemetryCollectionConsent":
+          settings.uploadTelemetry = message.consent;
```

Comment:
PingSender is initialized before this. You should call `pingSender._setUploadEnabled(false)`. There is a possible race condition with loading from prefs. Or maybe we don't need to sync with Firefox prefs anymore and can remove it? https://github.com/mozilla/firefox-translations/blob/7230c7f65c3b367d087b31b621bf5cfcbddd22da/extension/model/telemetry/PingSender.js#L48

Reply:
Let's keep the code to use the prefs setting but replace by the user consent for now in the case things change....again.